src/vos/vos_obj_cache.c

```diff
@@ -230,7 +230,9 @@ vos_obj_hold(struct daos_lru_cache *occ, daos_handle_t coh,
 
 		if ((!(obj->obj_df->vo_oi_attr & VOS_OI_PUNCHED) ||
 		     obj->obj_df->vo_latest >= epoch) &&
-		    obj->obj_df->vo_incarnation == obj->obj_incarnation) {
+		    obj->obj_df->vo_incarnation == obj->obj_incarnation &&
+		    !vos_dtx_obj_canceled(&cont->vc_pool->vp_umm,
```

Comment:
The object cache may bypass obj/key sharing, I have made some fixes in the patch PR230 for that. So you can ignore related fixes in your patch. Thanks.

Reply:
I can remove this here, but I am a bit concerned if you use vos_dtx_check, since the pmem read latency is still a bit higher than DRAM, and this is object cache.

"On average, the average idle read latency with Optane persistent memory is expected to be about 350 nanoseconds when applications direct the read operation to Optane persistent memory, or when the requested data is not cached in DRAM."

I believe DRAM is less than 20 nanoseconds.

Reply:
It is not easy to avoid the PMEM accessing for availability check because we have to handle shared update case unless we also cache DTX related logic, that will make things more complex. I prefer to resolve performance related things in next step.

Reply:
vos_dtx_obj_canceled() also needs to access PMEM.

Reply:
I am saying
1. we need object cache.
2. is there way to not always access PMEM to check canceled object.